lib/compiler/src/engine/artifact.rs

```diff
@@ -38,12 +37,13 @@ use wasmer_vm::{FunctionBodyPtr, MemoryStyle, TableStyle, VMSharedSignatureIndex
 use wasmer_vm::{InstanceAllocator, StoreObjects, TrapHandlerFn, VMConfig, VMExtern, VMInstance};
 
 pub struct AllocatedArtifact {
+    frame_info_registered: bool,
```

Comment:
Isn't a bool flag redundant if the registration itself is `Option`?

Reply:
No, because the FrameRegisteration is moved, and the Option is None again. Yet, the registration has been done and must not be done again. This is like that to avoid chnaging the API.